src/python/pants/backend/javascript/subsystems/nodejs.py

```diff
@@ -41,6 +45,28 @@ class NodeJS(TemplatedExternalTool):
         "linux_x86_64": "linux-x64",
     }
 
+    resolves = DictOption[str](
+        default={},
+        help=softwrap(
+            f"""
+            A mapping of names to lockfile paths used in your project.
+
+            Specifying a resolve name is optional. If unspecified,
+            the default resolve name is calculated by taking the path
+            from the build root to the directory containing the lockfile
+            and replacing '{os.path.sep}' with '.' in that path.
+
+            Example:
+            An npm lockfile located at `js/package/package-lock.json'
+            will result in a resolve named `js.package`.
```

Comment:
You might want to consider applying source roots here to allow people to strip common prefixes like `src`?